### PR TITLE
Implement old Argo test with new-e2e

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Migrate our last Argo e2e tests to new-e2e testing framework.

The last to migrate was k8s_e2e_cspm. The testing part from https://github.com/DataDog/datadog-agent/blob/b0d6c2d3373c39449f4589a5a0818f28b4f7df90/test/e2e/cws-tests/tests/test_e2e_cspm_kubernetes.py#L1
was moved to the same test defined in Go. All the setup part that was done with Argo is now handled by new-e2e framework.

### Motivation

Get rid of old testing frameworks

### CODE REVIEW

File: test/new-e2e/tests/cspm/cspm_test.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+		},
+	},
+}
+
+func TestCSPM(t *testing.T) {
+	values := `
+datadog:
+  securityAgent:
+    compliance:
+      enabled: true`
```
Comment (pducolin) [inline test/new-e2e/tests/cspm/cspm_test.go:175]: **💬 suggestion**
I prefer using go `embed` to avoid yaml formatting issues https://pkg.go.dev/embed

File: test/new-e2e/tests/cspm/cspm_test.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+	e2e.Run(t, &cspmTestSuite{}, e2e.WithProvisioner(localkubernetes.Provisioner(localkubernetes.WithAgentOptions(kubernetesagentparams.WithHelmValues(values), kubernetesagentparams.WithoutDualShipping()))))
```
Comment (pducolin) [inline test/new-e2e/tests/cspm/cspm_test.go:177]: **❓ question**
Should this use the `awskind` provisioner?

File: test/new-e2e/tests/cspm/cspm_test.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		metrics, err := s.Env().FakeIntake.Client().FilterMetrics("datadog.security_agent.compliance.containers_running")
+		assert.NoError(c, err)
+		assert.NotEmpty(c, metrics)
+		s.T().Log("Metrics found: datadog.security_agent.compliance.containers_running")
+	}, 2*time.Minute, 10*time.Second)
+
+}
+func (s *cspmTestSuite) checkFindings(findings, expectedFindings findings) {
+	checkedRule := []string{}
```
Comment (pducolin) [inline test/new-e2e/tests/cspm/cspm_test.go:216]: **💬 suggestion**
Would add `t.Helper()` here so that this logs with the caller rather than with this line

File: test/e2e/cws-tests/tests/test_e2e_cspm_docker.py [hunk 4]
```diff
@@ -1,150 +0,0 @@
...
-
-            self.docker_helper.wait_agent_container()
-
-        with Step(msg="check agent events", emoji=":check_mark_button:"):
-            self.container.exec_run("security-agent compliance check --dump-reports /tmp/reports.json --report")
-            _, output = self.container.exec_run("cat /tmp/reports.json")
-            print(output)
-            findings = json.loads(output)
-
-            expected_findings = {
```
Comment (pducolin) [inline test/e2e/cws-tests/tests/test_e2e_cspm_docker.py:62]: **❓ question**
Should you port also the docker test?

File: test/new-e2e/tests/cspm/cspm_test.go (hunk 2 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/cspm/cspm_test.go:177]: Oh yeah, used the local provisioner to write the test but still need to change that

**Review by YoannGh — APPROVED** [review]

**Review by safchain — APPROVED** [review]

**Review by liustanley — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

File: test/new-e2e/tests/cspm/cspm_test.go [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+func isSubset(a, b map[string]string) bool {
+	for k, v := range a {
+		if b[k] != v {
+			return false
+		}
+	}
+	return true
+}
```
Comment (L3n41c) [inline test/new-e2e/tests/cspm/cspm_test.go:255]: We need to differentiate between an element which is absent from the map and an element which is present with an empty value.

The current implementation would return `true` for `isSubset(map[string]string{"foo":""}, map[string]string{})` which is weird because a non-empty map cannot be a subset of the empty map.

```suggestion
func isSubset(a, b map[string]string) bool {
	for k, v := range a {
		if vb, found := b[k]; !found || vb != v {
			return false
		}
	}
	return true
}
```

**Review by amenasria — APPROVED** [review]
LGTM for Agent DevX Infra owned files !

File: test/new-e2e/tests/cspm/cspm_test.go (hunk 5 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/cspm/cspm_test.go:255]: Oh yeah, did not think about that edge case and Go defaulting non-present key to default value for the type. Applied your suggestion!

**Review by pducolin — APPROVED** [review]
